.github/workflows/test.yml

```diff
@@ -340,70 +102,120 @@ jobs:
-              run: AZLE_PROPTEST_NUM_RUNS=10 AZLE_PROPTEST_VERBOSE=true npm test
-            - if: ${{ needs.release-candidate-deploy.outputs.should_run_tests && (github.ref == 'refs/heads/main' && contains(github.event.head_commit.message, 'Merge pull request') && contains(github.event.head_commit.message, 'demergent-labs/release--')) }}
+
+            - name: Run Test
+              run: |
+                  BRANCH_NAME="${{ github.head_ref }}"
+                  MAIN_BRANCH='refs/heads/main'
+                  COMMIT_MESSAGE="${{ github.event.head_commit.message }}"
+                  RELEASE_TAG='release--'
+                  RELEASE_PR_MERGE_MSG='Merge pull request'
+                  RELEASE_REPO_MSG='demergent-labs/release--'
+
+                  IS_FEATURE_PR=true
+                  IS_RELEASE_PR=false
+                  IS_MERGE_TO_MAIN_FROM_RELEASE=false
+
+                  # Check if it's a feature pull request
+                  if [[ $BRANCH_NAME != *"$RELEASE_TAG"* && ${{ github.ref }} != "$MAIN_BRANCH" && !($COMMIT_MESSAGE == *"$RELEASE_PR_MERGE_MSG"* && $COMMIT_MESSAGE == *"$RELEASE_REPO_MSG"*) ]]; then
+                      IS_FEATURE_PR=true
+                  else
+                      IS_FEATURE_PR=false
+                  fi
+
+                  # Check if it's a release candidate pull request
+                  if [[ $BRANCH_NAME == *"$RELEASE_TAG"* && ${{ github.ref }} != "$MAIN_BRANCH" && !($COMMIT_MESSAGE == *"$RELEASE_PR_MERGE_MSG"* && $COMMIT_MESSAGE == *"$RELEASE_REPO_MSG"*) ]]; then
+                      IS_RELEASE_PR=true
+                  else
+                      IS_RELEASE_PR=false
+                  fi
+
+                  # Check if it's a merge from a release candidate into main
+                  if [[ ${{ github.ref }} == "$MAIN_BRANCH" && $COMMIT_MESSAGE == *"$RELEASE_PR_MERGE_MSG"* && $COMMIT_MESSAGE == *"$RELEASE_REPO_MSG"* ]]; then
+                      IS_MERGE_TO_MAIN_FROM_RELEASE=true
+                  else
+                      IS_MERGE_TO_MAIN_FROM_RELEASE=false
+                  fi
+
+                  # Determine the number of runs based on the conditions
+                  if [[ $IS_MERGE_TO_MAIN_FROM_RELEASE == true ]]; then
+                      RUNS=100
+                  elif [[ $IS_RELEASE_PR == true ]]; then
+                      RUNS=10
+                  else
+                      RUNS=5
+                  fi
+
+                  AZLE_PROPTEST_NUM_RUNS=$RUNS AZLE_PROPTEST_VERBOSE=true npm test
               shell: bash -l {0}
-              working-directory: ${{ matrix.example_directories }}
-              run: AZLE_PROPTEST_NUM_RUNS=100 AZLE_PROPTEST_VERBOSE=true npm test
+              working-directory: ${{ matrix.tests.path }}
 
-    check-basic-integration-tests-success:
-        needs: basic-integration-tests
+    # These final jobs are designed to ensure that all jobs spun up from the matrix in the run-test have succeeded
+    check-basic-integration-test-success:
```

Comment:
Should we change these to `check-test-success` or something else? I know I will have to change them in the repo settings.

Reply:
yes, that task is already mentioned above, I was holding off until last minute because it was making my tests fail since the repo settings weren't updated yet